### PR TITLE
Add reward field in validators table

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #2135

## Implementation details
1. Add current reward and current ratio field
2. Add message update reward
3. Add handler reward

Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: CHANGELOG_UNRELEASED.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 ### Chain
 
+- (impv) [\#2135](https://github.com/bandprotocol/bandchain/pull/2135) Add reward field in validators table.
```
Comment (taobun) [inline CHANGELOG_UNRELEASED.md]: Wrong pull request number

File: flusher/flusher/handler.py [hunk 2]
```diff
@@ -67,3 +67,9 @@ def handle_set_validator(self, msg):
             .values(**msg)
             .on_conflict_do_update(constraint="validators_pkey", set_=msg)
         )
+
+    def handle_update_validator_reward(self, msg):
```
Comment (taobun) [inline flusher/flusher/handler.py]: handle_update_validator

**Review by taobun — CHANGES_REQUESTED** [review]

File: CHANGELOG_UNRELEASED.md [hunk 3]
```diff
@@ -12,6 +12,8 @@
 
 - (impv) [\#2132](https://github.com/bandprotocol/bandchain/pull/2132) Implement emitter handler for bank messages.
 - (patch) [\#2135](https://github.com/bandprotocol/bandchain/pull/2135) Bump Cosmos-SDK to version 0.38.5.
+- (impv) [\#2135](https://github.com/bandprotocol/bandchain/pull/2135) Add reward field in validators table.
+- (impv) [\#2136](https://github.com/bandprotocol/bandchain/pull/2136) Add reward field in validators table.
```
Comment (sorawit) [inline CHANGELOG_UNRELEASED.md:16]: Please remove the duplicated line

File: chain/emitter/distribution.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	dist "github.com/cosmos/cosmos-sdk/x/distribution"
+)
+
+// handleMsgWithdrawDelegatorReward implements emitter handler for MsgWithdrawDelegatorReward.
+func (app *App) handleMsgWithdrawDelegatorReward(
+	txHash []byte, msg dist.MsgWithdrawDelegatorReward, evMap EvMap, extra JsDict,
+) {
+	app.emitUpdateValidatorReward(msg.ValidatorAddress)
```
Comment (sorawit) [inline chain/emitter/distribution.go:39]: Add the beneficiary to related accounts

File: chain/emitter/staking.go [hunk 5]
```diff
@@ -2,11 +2,37 @@ package emitter
...
+
+func (app *App) getCurrentRatio(addrs sdk.ValAddress, reward types.ValidatorCurrentRewards) string {
+	latestReward := app.DistrKeeper.GetValidatorHistoricalRewards(app.DeliverContext, addrs, reward.Period-1)
+	if !latestReward.CumulativeRewardRatio.IsZero() {
+		return latestReward.CumulativeRewardRatio[0].Amount.String()
+	}
+	return "0"
+}
+
+func (app *App) emitUpdateValidatorReward(addrs sdk.ValAddress) {
```
Comment (sorawit) [inline chain/emitter/staking.go]: This is singular? `addrs`. Dtto for all places.

File: chain/emitter/staking.go [hunk 6]
```diff
@@ -2,11 +2,37 @@ package emitter
...
+	app.Write("UPDATE_VALIDATOR", JsDict{
+		"operator_address": addrs.String(),
+		"current_reward":   getCurrentReward(reward),
+		"current_ratio":    app.getCurrentRatio(addrs, reward),
+	})
+}
+
 func (app *App) emitSetValidator(addrs sdk.ValAddress) {
 	val, _ := app.StakingKeeper.GetValidator(app.DeliverContext, addrs)
+	reward := app.DistrKeeper.GetValidatorCurrentRewards(app.DeliverContext, addrs)
```
Comment (sorawit) [inline chain/emitter/staking.go]: Let's move this to be part of `getCurrentReward`

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/emitter/staking.go [hunk 7]
```diff
@@ -2,11 +2,37 @@ package emitter
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
 	"github.com/cosmos/cosmos-sdk/x/staking"
 )
 
+func getCurrentReward(reward types.ValidatorCurrentRewards) string {
```
Comment (sorawit) [inline chain/emitter/staking.go]: Can we move these two functions to `distribution.go`?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**